src/main/js/controls/Bubble/helpers/helpers.js

```diff
@@ -0,0 +1,559 @@
+ * Transforms the point in the Line graph on resize
+ *
+ * @private
+ * @param {object} scale - d3 scale for Graph
+ * @returns {Function} - translate function for d3 transform
+ */
+const transformPoint = (scale) => (value) => {
+    const getX = (val) => scale.x(val.x);
+    const getY = (val) => scale[val.yAxis](val.y);
+    return `translate(${getX(value)},${getY(value)})`;
+};
+/**
+ * Transforms points for a data point set in the Line graph on resize
+ *
+ * @private
+ * @param {object} scale - d3 scale for Graph
+ * @param {d3.selection} canvasSVG - d3 selection node of canvas svg
+ * @param {string} cls - selector for the data point translation
+ * @param {object} config - config object derived from input JSON
+ * @returns {object} d3 select object
+ */
+const translatePoints = (scale, canvasSVG, cls, config) =>
+    canvasSVG
+        .selectAll(`.${styles.bubbleGraphContent} .${cls}`)
+        .each(function(d) {
+            const pointSVG = d3.select(this);
+            pointSVG
+                .transition()
+                .call(
+                    constants.d3Transition(config.settingsDictionary.transition)
+                )
+                .attr("transform", () => transformPoint(scale)(d));
+        });
+
+/**
+ * Toggles the selection of a data point, executes on click of a data point.
+ *
+ * @private
+ * @param {object} target - DOM element of the data point clicked
+ * @returns {Array} d3 html element of the selected point
+ */
+const toggleDataPointSelection = (target) => {
+    blurActionHandler(target, constants.HOVER_EVENT.MOUSE_ENTER);
+    const selectedPointNode = d3
+        .select(target.parentNode)
+        .select(`.${styles.dataPointSelection}`);
+    selectedPointNode.attr(
+        "aria-hidden",
+        !(selectedPointNode.attr("aria-hidden") === "true")
+    );
+    return selectedPointNode;
+};
+
+/**
+ * Handler for the data point on click. If the content property is present for the data point
+ * then the callback is executed other wise it is NOP.
+ * If the callback is present, the selected data point is toggled and the element is passed as an argument to the
+ * consumer in the callback, to execute once the popup is closed.
+ *  Callback arguments:
+ *      Post close callback function
+ *      value [x and y data point values]
+ *      Selected data point target [d3 target]
+ *  On close of popup, call -> the provided callback
+ *
+ * @private
+ * @param {object} value - data point object
+ * @param {number} index - data point index for the set
+ * @param {object} target - DOM object of the clicked point
+ * @returns {undefined} - returns nothing
+ */
+const dataPointActionHandler = (value, index, target) => {
+    if (utils.isEmpty(value.onClick)) {
+        return;
+    }
+    toggleDataPointSelection(target).call((selectedTarget) =>
+        value.onClick(
+            () => {
+                selectedTarget.attr("aria-hidden", true);
+                removeBubbleBlur(getAllCircleNodes());
```

Comment:
Same here.
> Instead of doing this, can we just blur all of them and just unblur the selected one? This will save all the preprocessing in getAllCircleNodes.

Reply:
This is pending.

Reply:
I will remove this getAllCircleNodes